### ISSUE TITLE
Add support for ALTCHA

### ISSUE THREAD

**Issue by mfelczak** [pkp/pkp-lib#10283]
[ALTCHA](https://altcha.org/) is a free, open source, and self-hosted alternative to reCaptcha. It's also more accessible since it doesn't require any user interaction. Github repo: https://github.com/altcha-org/altcha

![Screen Shot 2024-08-06 at 3 33 39 PM](https://github.com/user-attachments/assets/a7a7f27d-bed2-4af7-b463-84cedb041e6e)

**Comment by mfelczak** [comment]
Hello @asmecher, before we start implementation I wanted to confirm recommended approach with you: should this be modelled on the existing reCaptcha implementation with install-wide settings in config.inc.php and a corresponding FormValidator subclass?

**Comment by asmecher** [comment]
Hi Michael -- Yes, on balance I think that's best. If we are going to have several options then in my wildest dreams we'd have an interface that all options could meet, but if it's just two options (ALTCHA and ReCAPTCHA) then there's not enough benefit to justify the overhead in coding.

**Comment by Godoy0722** [comment]
I just implemented ALTCHA integration for login, register and reset password pages. You can see it on [#10456](https://github.com/pkp/pkp-lib/pull/10456).

For this support to work, the `config.inc.php` must be updated with the new variables as I also created PRs as follows:
 - [OPS/775](https://github.com/pkp/ops/pull/775)
 - [OJS/4446](https://github.com/pkp/ojs/pull/4446)
 - [OMP/1703](https://github.com/pkp/omp/pull/1703)


**To Reproduce**
Steps to reproduce the behavior:
1. Starting on the root directory of any system, Run `composer --working-dir=lib/pkp install`
2. Update the `config.inc.php` file with the new variables as shown on the other PRs, and set `altcha` and `altcha_on_login` as `on`
3. Enter on login page and click on submit button

**Comment by Godoy0722** [comment]
@asmecher just a heads-up that the PRs are ready to be reviewed!

**Comment by asmecher** [comment]
Apologies for the long wait, @Godoy0722 -- I've added some feedback [here](https://github.com/pkp/pkp-lib/pull/10456#pullrequestreview-2450139395).

**Comment by Godoy0722** [comment]
Hello @asmecher . Just to let you know, I implemented the required updates if you could please take a look. I also added the altcha as a dependency for all OJS, OMP, and OPS so it can work without the minified file I added before.

**Comment by asmecher** [comment]
Thanks, @Godoy0722, I've got all the PRs merged (manually)!

**Comment by asmecher** [comment]
@Tribunal33, this would be a good one to get your testing eyes on.

**Comment by Tribunal33** [comment]
@asmecher I was able to get it up and running on my VM.  I couldn't find a way to generate a failed validation, but this could be its frictionless intent.  Besides this it looks to be working correctly.  I created a small space for it in the 3.5 release area in QA zone.  

Test Case Overview

1. Test login Valid
2. Test registration is Valid
3. Test forgot password is Valid
4. Test login, registration or forgot password warning message
   1. load up the particular page (login, registration or forgot password)
   2. Then you will need to change the encrypt config key manually to something bigger than its current value.
   3. This will cause the error to be seen
      1.

      ![Validation message](.attachments.40160/image.png)
5. Could not find a way to generate a failed validation (working on finding out how)

**Comment by asmecher** [comment]
@Godoy0722 Is there a way that you know of to intentionally cause the APTCHA to fail? For testing purposes for @Tribunal33.

**Comment by Godoy0722** [comment]
I just searched in the ALTCHA docs, and I found a property to make a mocked error inside the ALTCHA widget, which is to pass `mockerror` property into it. I'm attaching the [Docs](https://altcha.org/docs/website-integration/#configuration) I looked for.

Example for testing:

```
<altcha-widget ... mockerror></altcha-widget>
```

**Comment by Tribunal33** [comment]
Thank you @Godoy0722 I have updated the documents to reflect these possible error messages.  This has been confirmed and tested as well.
